test/arguments.test.js

```diff
@@ -1,135 +1,120 @@
 const loader = require('../arguments')
 
-const parser = new Parser()
+const dummyLoaderRegistry = {
+  load: () => undefined
+}
 
 describe('arguments loader', () => {
   const term = rdf.namedNode('http://example.com/')
-  let loaderRegistry
-
-  beforeEach(() => {
-    loaderRegistry = {
-      load: sinon.stub()
-    }
-  })
 
   describe('loading from rdf:List', () => {
-    test('should fall back to verbatim literal value', async () => {
-      // given
+    it('should fall back to verbatim literal value', async () => {
       const dataset = await loadDataset('./arguments-list.ttl')
 
-      // when
-      const result = await loader({ term, dataset }, { loaderRegistry })
+      const result = await loader({ term, dataset }, { loaderRegistry: dummyLoaderRegistry })
 
-      // then
-      expect(result).toEqual(['a', '5'])
+      deepStrictEqual(result, ['a', '5'])
     })
 
-    test('should handle boolean false, but not undefined values from loaders', async () => {
-      // given
+    it('should handle boolean false, but not undefined values from loaders', async () => {
       const dataset = await loadDataset('./arguments-list.ttl')
       const values = [undefined, '']
       const loaderRegistry = {
         load: () => values.shift()
       }
 
-      // when
       const result = await loader({ term, dataset }, { loaderRegistry })
 
-      // then
-      expect(result).toEqual(['a', ''])
+      deepStrictEqual(result, ['a', ''])
     })
 
-    test('should use loaders to load values', async () => {
-      // given
+    it('should use loaders to load values', async () => {
       const dataset = await loadDataset('./arguments-list.ttl')
-      loaderRegistry.load.callsFake((arg) => arg.value.toUpperCase())
+      const loaderRegistry = {
+        load: ptr => ptr.value.toUpperCase()
+      }
 
-      // when
       const result = await loader({ term, dataset }, { loaderRegistry })
 
-      // then
-      expect(result).toEqual(['A', '5'])
+      deepStrictEqual(result, ['A', '5'])
     })
 
-    test('should forward options to loaderRegistry', async () => {
-      // given
+    it('should forward options to loaderRegistry', async () => {
+      let called = null
+
+      const loaderRegistry = {
+        load: (ptr, options) => {
+          called = options
+        }
+      }
+
       const options = {
         loaderRegistry,
         context: {},
         variables: new Map(),
         basePath: '/some/path'
       }
+
       const dataset = await loadDataset('./arguments-list.ttl')
 
-      // when
       await loader({ term, dataset }, options)
 
-      // then
-      expect(loaderRegistry.load.calledWith(
-        sinon.match.object,
-        {
-          context: options.context,
-          variables: options.variables,
-          basePath: options.basePath
-        }
-      )).toBe(true)
+      for (const key of ['basePath', 'context', 'variable']) {
+        strictEqual(called[key], options[key])
+      }
```

Comment:
I don't see how this is "cleaner" than using sinon.

Reply:
I think that was the only place the sinon code was a little bit more readable as the plain js code. E.g. I even consulted the docs for [this line](https://github.com/zazuko/rdf-loader-code/pull/18/files#diff-9b26909759830130c077d1cdf3542f68L19), but it doesn't describe what the default stub does.